### PR TITLE
lib: add qualifiers to pre- and postconditions

### LINKED ISSUE #3410: `pre`- and `post`-condition qualification missing in base lib
I had to make this change to while debugging DFA analysis for effects (I expected my code not to cause any pre/postfault if debug and safety is turned off):
```
> git diff lib/integer.fz 
diff --git a/lib/integer.fz b/lib/integer.fz
index dbd028f75..213b2cd9f 100644
--- a/lib/integer.fz
+++ b/lib/integer.fz
@@ -46,7 +46,7 @@ module:public integer : numeric is
   # bitwise NOT
   public prefix ~ integer.this
     pre
-      is_bounded
+      debug: is_bounded
   => abstract
 
   # bitwise NOT (Unicode alias)
@@ -107,7 +107,8 @@ module:public integer : numeric is
 
     # this digit as an UTF-8 byte
     digit_as_utf8_byte(d u8) u8
-    pre d ≤ 35 # 35 would be a Z
+    pre
+      debug: d ≤ 35 # 35 would be a 'Z'
     =>
       if d < 10 then String.zero_char + d else String.a_char + d - 10
```
Similar qualifiers `debug` and `safety` are missing in other features in the base lib and should be added. As a general guideline: `debug` if the result is undefined or wrong but harmless, `safety` if a crash/segfault/illegal memory access/etc. would be caused in case the condition is not respected.

### PR DESCRIPTION
**Pull request by simonvonhackewitz**
unsure about some of them, e.g.
* int.fz as_uXX
* integer.fz prefix ~
* ...

resolve #3410

### CODE REVIEW

File: lib/container/Circular_Buffer.fz [hunk 1]
```diff
@@ -82,7 +82,7 @@ is
   #
   public flush (n i64) array T
   pre
-    n ≤ buffered
+    safety: n ≤ buffered
```
Comment (michaellilltokiwa) [inline lib/container/Circular_Buffer.fz:85]: Also debug

File: lib/container/Set.fz [hunk 2]
```diff
@@ -127,8 +127,8 @@ public Set(public E type : property.equatable) ref : Sequence E is
   # equality
   #
   public fixed type.equality(a, b container.Set E) bool
-  pre a.finite
-      b.finite
+  pre safety: a.finite
```
Comment (michaellilltokiwa) [inline lib/container/Set.fz:130]: debug

File: lib/io/buffered/reader.fz [hunk 3]
```diff
@@ -76,9 +76,9 @@ public reader(LM type : mutate, rp Read_Provider, buf_size i32) : effect effect_
   # fill the currently empty buffer with up to max_n bytes
   #
   fill_buffer(max_n i32) unit
-  pre match buffer.get
-        a array => a.is_empty
-        io.end_of_file => false
+  pre safety: match buffer.get
```
Comment (michaellilltokiwa) [inline lib/io/buffered/reader.fz:79]: These here also, rather debug than safety, I think?

File: lib/process/spawn.fz [hunk 4]
```diff
@@ -90,7 +90,7 @@ private:public Process(id i64, std_in option i64, std_out, std_err i64) ref is
   # helper function
   #
   write_bytes (desc i64, bytes Sequence u8) outcome i32
-  pre bytes.count > 0
+  pre safety: bytes.count > 0
```
Comment (michaellilltokiwa) [inline lib/process/spawn.fz:93]: debug

**Review by michaellilltokiwa — CHANGES_REQUESTED** [review]

**Comment by michaellilltokiwa** [comment]
@simonvonhackewitz There are now conflicts unfortunately... Can you merge main, please?

File: lib/io/buffered/reader.fz (hunk 3 above)
Comment (fridis) [reply lib/io/buffered/reader.fz:79]: `safety` should IMHO really protect the system from a crash, so I would go for `debug` here. The same for the following 2x`safety`

File: lib/uint.fz [hunk 5]
```diff
@@ -110,7 +110,7 @@ is
   # returns the quotient and the remainder
   # NYI performance: https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/math/big/natdiv.go
   fixed divide_with_remainder (divisor uint) tuple uint uint
-  pre divisor > uint.zero
+  pre safety: divisor > uint.zero
```
Comment (fridis) [inline lib/uint.fz:113]: here `safety` is probably justified, the code might run into an recursion or loop if we divide by zero.

**Review by fridis — CHANGES_REQUESTED** [review]
Just minor clarifications on `safety` vs `debug`.

**Review by michaellilltokiwa — APPROVED** [review]

**Pull request merged**